openmc/material.py

```diff
@@ -914,6 +915,97 @@ def to_xml_element(self, cross_sections=None):
 
         return element
 
+    @classmethod
+    def mix_materials(cls, materials, fracs, percent_type='ao', name=None):
+        """Mix materials together based on atom, weight, or volume fractions
+
+        Parameters
+        ----------
+        materials : Iterable of openmc.Material
+            Materials to combine
+        fracs : Iterable of float
+            Fractions of each material to be combined
+        percent_type : str
+            Type of percentage, must be one of 'ao', 'wo', or 'vo', to signify atom
+            percent (molar percent), weight percent, or volume percent, 
+            optional. Defaults to 'ao'
+        name : str
+            The name for the new material, optional. Defaults to concatenated
+            names of input materials with percentages indicated inside
+            parentheses.
+
+        Returns
+        -------
+        new_mat : openmc.Material instance for the new mixture
+
+        """
+
+        cv.check_type('materials', materials, Iterable, Material)
+        cv.check_type('fracs', fracs, Iterable, Real)
+        cv.check_value('percent type', percent_type, {'ao', 'wo', 'vo'})
+
+        fracs = np.asarray(fracs)
+        void_frac = 1. - np.sum(fracs)
+
+        # Warn that fractions don't add to 1, set remainder to void, or raise
+        # an error if percent_type isn't 'vo'
+        if not np.isclose(void_frac, 0.):
+            if percent_type in ['ao', 'wo']:
+                msg = 'A non-zero void fraction is not acceptable for \
+                    percent_type: {}'.format(percent_type)
+                raise ValueError(msg)
+            else:
+                msg = 'Warning: sum of fractions do not add to 1, void fraction \
+                    set to {}'.format(void_frac)
+                warnings.warn(msg)
+
+        # Calculate appropriate weights which are how many cc's of each
+        # material are found in 1cc of the composite material
+        amms = np.asarray([mat.average_molar_mass for mat in materials])
+        mass_dens = np.asarray([mat.get_mass_density() for mat in materials])
+        if percent_type == 'ao':
+            wgts = fracs * amms / mass_dens
+            wgts /= np.sum(wgts)
+        elif percent_type == 'wo':
+            wgts = fracs / mass_dens
+            wgts /= np.sum(wgts)
+        elif percent_type == 'vo':
+            wgts = fracs
+
+        # If any of the involved materials contain S(a,b) tables raise an error
+        sab_names = set(sab[0] for mat in materials for sab in mat._sab)
+        if sab_names:
+            msg = 'Currently we do not support mixing materials containing \
+                S(a,b) tables'
```

Comment:
This will result in a bunch of extra whitespace in the middle of the string. Better to just use implicit concatenation with something like:
```Python
msg = ('some very long string '
       'that continues over multiple lines')
```
When Python sees two string literals adjacent to one another, it joins them together